client/lib/signup/step-actions/index.js

```diff
@@ -39,6 +43,22 @@ import { getSiteId } from 'calypso/state/sites/selectors';
 const Visibility = Site.Visibility;
 const debug = debugFactory( 'calypso:signup:step-actions' );
 
+const isUserAssignedFSEBeta = ( userId ) => {
+	const fse_beta_rollout_percentage_start_flow = 10;
+
+	if ( ! userId ) {
+		return false;
+	}
```

Comment:
Will this ever be the case? In what cases will someone that is not logged in be able to create a site?

Reply:
No, not given the way onboarding currently works. We know this because we happen to know the function is only called from within `createSiteWithCart()`. But I'm imagining `isUserAssignedFSEBeta` is a useful lib function that may be pulled out into a common place before this whole FSE rollout project is through. So I think it's better to be defensive.

Reply:
Actually, this is _can_ happen. If you test the flow with a brand new user (in incognito mode for example) then `userId === null`

Reply:
> If you test the flow with a brand new user (in incognito mode for example) then userId === null

This rollout is primarily intended for brand new users, so we should change this to make sure they are included too. If there are no better options, instead of `userId` here we could generate a random number between 0 and 99.

Reply:
Ideally, we'd have something deterministic because we want to change other parts of their experience based on their eligibility i.e. only showing non-universal block themes to eligible users. We could use a random number here and then use the `/block-editor` endpoint whenever we wanted to check their availability later on. But my idea here https://github.com/Automattic/wp-calypso/pull/60330#issuecomment-1019666501 makes the ID of brand new users available in the required place.